src/lib/optimizer/optimizer.cpp

```diff
@@ -112,6 +113,10 @@ std::shared_ptr<Optimizer> Optimizer::create_default_optimizer() {
   // Bring predicates into the desired order once the PredicatePlacementRule has positioned them as desired
   optimizer->add_rule(std::make_unique<PredicateReorderingRule>());
 
+  // Before the IN predicate is rewritten, it should have been moved to a good position. Also, it should not take
+  // part in the join ordering even if it is rewritten as a join, so we run this rule close to the end.
```

Comment:
```
  // Before the IN predicate is rewritten, it should have been moved to a good position. Also, while the IN predicate
  // might become a join, it is semantically more similar to a predicate. If we run this rule too early, it might
  // hinder other optimizations that stop at joins. For example, the join ordering currently does not know about semi
  // joins and would not recognize such a rewritten predicate.
```
Better?

Reply:
!